spark-indexer/src/test/java/org/uniprot/store/spark/indexer/suggest/SuggestDocumentsToHDFSWriterTest.java

```diff
@@ -148,6 +149,37 @@ void getKeyword() {
         assertEquals("Coding sequence diversity", resultMap.get("KW-9997").get(0).value);
     }
 
+    @Test
+    void getProteome() {
+        SuggestDocumentsToHDFSWriter writer = new SuggestDocumentsToHDFSWriter(parameter);
+        JavaRDD<SuggestDocument> suggestRdd = writer.getProteome();
+        assertNotNull(suggestRdd);
+        int count = (int) suggestRdd.count();
+        assertEquals(6, count);
+
+        Map<String, List<SuggestDocument>> resultMap =
+                getResultMap(suggestRdd.take(count), doc -> doc.id);
+
+        assertThat(resultMap.containsKey("UP000008687"), is(true));
+        assertThat(resultMap.get("UP000008687").size(), is(1));
+        assertNotNull(resultMap.get("UP000008687").get(0));
+        assertEquals(PROTEOME_UPID.name(), resultMap.get("UP000008687").get(0).dictionary);
+        assertEquals("UP000008687", resultMap.get("UP000008687").get(0).id);
+        assertEquals("Potato virus X (strain X3) (PVX)", resultMap.get("UP000008687").get(0).value);
+        assertEquals(1, resultMap.get("UP000008687").get(0).altValues.size());
+        assertEquals("UP000008687", resultMap.get("UP000008687").get(0).altValues.get(0));
+    }
+
+    @Disabled
```

Comment:
I had to remove the test case because the the feature of Oracle like `CONNECT BY PRIOR` is not supported by H2. It will not be easy to convert the recursive query equivalent in H2. 
SuggestDocumentsToHDFSWriter.getUniParcTaxonomy is basically calling TaxonomyRDDReader.
And we have already covered basic tests in TaxonomyRDDReaderTest for TaxonomyRDDReader.
Let me know what do you think?

Reply:
looking at how it works in Spring they solved some H2 not supported problem as in, `TaxonomyJobIT`, where they override the actual query. do you think something like this would work here for Spark? If not no worries.